### PR TITLE
GitHub Cleanup

### PR DESCRIPTION
**Pull request by LRWeber**
- Improved `github/queryManager` code.
  - No longer uses `subprocess` library, replaced with `requests` for a more OS agnostic solution.
  - Added option to specify line endings for `fileSave`, more cross-OS project friendly.
  - General cleanup and simplification.
- Fixed `README` code blocks.
  - Standardized code block style to fenced (vs indented).
  - Added `shell` syntax highlighting
  - Changed `json` blocks to `jsonc` for appropriate comment highlighting when displayed on GitHub. (No more bright red boxes!)
- Fixed typos in `README` and `github/__init__.py`.
- Added `.vscode` to `.gitignore`

- Added instructions to `README` for installation from PyPI

### CODE REVIEW

**Comment by leebrian** [comment]
Neat, thanks for this cleanup, I especially think the readme changes will help new users. 

It looks like some of your files are changing single quotes to doubles. When diffing it takes a little more time to filter out the lines with real changes vs the ones that change quotes.

If you think it’s valuable to use doubles, could you maybe send a pr to change the entire project. I don’t know Ian’s preference as that’s probably most important. 

I think using the same quotes throughout the project will help with maintainability and to attract more great contributors.

**Comment by LRWeber** [comment]
I don't really have a personal preference one way or the other for singles vs doubles. That file, as it was, had a mixing of both. I changed to doubles because it looked like they were the majority. However, I see now that other Python files in the repo tend to favor single quotes!

I can certainly make the change the other direction. I agree that there should be consistency across the project. Though I'll similarly defer to @IanLee1521 for recommendations as to what that style should look like.

**Comment by IanLee1521** [comment]
I think part of this is that we at LLNL started playing with using Black as our linter...  I think that is what we should do, but I agree that it would be good to split that bit out to a separate PR and do it on everything.

@LRWeber , can you do that?

**Comment by IanLee1521** [comment]
And if we merge that first, it should clear up some of the issues here in this pull request that confuse the differences.

**Comment by LRWeber** [comment]
Agreed, that will simplify the cleanup (and future maintenance).
I'll work on that now!

**Comment by IanLee1521** [comment]
🎉

**Comment by leebrian** [comment]
That will be cool. I think CI runs flake8 for peps and whatnot. Can it also run Black to check for future checkins?

**Comment by LRWeber** [comment]
It may be worth noting that if both are running, we may have to add some exclusions to our flake8 options. For the most part, code that clears black should be cleared by flake8's default formatting rules as well, but I've seen a few exceptions. (e.g. flake8 W503/W504 line break before/after binary operator)

**Comment by LRWeber** [comment]
If it's an automatic check, we may also want to run it as `black --check .`. Otherwise, it will modify the files by default. It may be better to return success/failure like flake8 currently does rather than making changes out of our control.

**Comment by IanLee1521** [comment]
@LRWeber -- Since this is all your code, do you mind taking a look at the conflict and resolving when able, now that I merged #47 ?

**Comment by LRWeber** [comment]
@IanLee1521 -- Conflicts should all be resolved, and I added a few more improvements.

File: README.md [hunk 1]
```diff
@@ -14,102 +14,110 @@ library can help with that!
 To get started, you will need a [GitHub Personal Auth
 Token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
 to make requests to the GitHub API. This should be set in your environment or
-shell ``rc`` file with the name ``GITHUB_API_TOKEN``:
```
Comment (IanLee1521) [inline README.md]: Where is this markdown formatting coming from? Are you using a tool for doing this linting?

File: README.md [hunk 2]
```diff
@@ -14,102 +14,110 @@ library can help with that!
...
 
 The configuration file is a json file that specifies what repository platforms
-to pull projects from  as well as some settings that can be used to override
+to pull projects from as well as some settings that can be used to override
 incomplete or inaccurate data returned via the scraping.
 
 The basic structure is:
 
-```json
+```jsonc
```
Comment (IanLee1521) [inline README.md]: Nice! Is `.jsonc` an actual file type / flavor that parsers know how to handle? I know some are very picky...

Comment (LRWeber) [reply README.md]: It seems to be relatively common. At worst, I think most markdown parsers tend to quietly overlook any unknown syntax specifications and just display the mono-spaced code block without any special highlighting. I think that's still an improvement over the angry red error highlights from the invalid json.

File: README.md (hunk 1 above)
Comment (LRWeber) [reply README.md]: Yes, I've been using most of the recommendations from [vscode-markdownlint](https://github.com/DavidAnson/vscode-markdownlint).

Comment (IanLee1521) [reply README.md]: any way to do this in travis?

File: README.md (hunk 2 above)
Comment (IanLee1521) [reply README.md]: agreed!

File: README.md (hunk 1 above)
Comment (LRWeber) [reply README.md]: I'm not particularly well versed in Travis yet. However, I know that the VS Code extension is powered by markdownlint for Node.js. If we can use the npm package in Travis, we'll have the equivalent markdownlint command line tool.

**Pull request merged**